web/src/message_events.js

```diff
@@ -86,6 +82,14 @@ export function insert_new_messages(messages, sent_by_this_client, deliver_local
         }
     }
 
+    for (const msg_list_data of message_lists.non_rendered_data()) {
+        if (!msg_list_data.filter.can_apply_locally()) {
+            message_list_data_cache.remove(msg_list_data.filter);
+        } else {
+            msg_list_data.add_messages(messages);
+        }
+    }
```

Comment:
Hmm, the previous logic does some extra work beyond `add_messages` that seems lost here?

``` ts
export function add_new_messages(                                                                       
    messages: Message[],                                                                                
    msg_list: MessageList,                                                                              
): RenderInfo | undefined {                                                                             
    if (!msg_list.data.fetch_status.has_found_newest()) {                                               
        // We don't render newly received messages for the message list,                                
        // if we haven't found the latest messages to be displayed in the                               
        // narrow. Otherwise the new message would be rendered just after                               
        // the previously fetched messages when that's inaccurate.                                      
        msg_list.data.fetch_status.update_expected_max_message_id(messages);                            
        return undefined;                                                                               
    }                                                                                                   
    return add_messages(messages, msg_list, {messages_are_new: true});                                  
}                                                                                                       
```

Reply:
Ooops! Forgot to update it.

Reply:
Updated!